pkg/loadbalancer/experimental/healthserver.go

```diff
@@ -183,13 +186,30 @@ func (s *healthServer) controlLoop(ctx context.Context, health cell.Health) erro
 					ServicePort: port,
 				},
 			)
+
+			// Find NodePort addr to use as a backend for $LB_VIP:$HC_NODEPORT frontend.
+			beAddr := netip.IPv4Unspecified()
+			is4 := fe.Address.AddrCluster.Is4()
+			if !is4 {
+				beAddr = netip.IPv6Unspecified()
+			}
+			for addr := range s.params.NodeAddresses.List(wtxn, tables.NodeAddressNodePortIndex.Query(true)) {
```

Comment:
This won't work since the address we pick may go away. We would need to either reconcile the changes or we could go via the NodeIP in the local node (via LocalNodeStore) as that's somewhat guaranteed not to change. Or if we can make the loopback address work that'd be even better.

Reply:
Note to myself: ideally use https://github.com/cilium/cilium/pull/32144